src/Data/Cursor/AbstractCursor.php

```diff
@@ -134,6 +144,99 @@ protected function is_valid_offset_and_node() {
 		return true;
 	}
 
+	/**
+	 * Validates threshold field configuration. Validation failure results in a fatal
+	 * error because query execution is guaranteed to fail.
+	 *
+	 * @param array $field  Threshold configuration.
+	 * 
+	 * @throws InvariationViolation Invalid configuration format.
+	 * 
+	 * @return void
+	 */
+	protected function is_valid_threshold_field( $field ) {
+		$class = get_called_class();
```

Comment:
We only need the $class for the errors it seems. Let's `sprintf() it into the error instead of calling it here.